### PR TITLE
Improve Contributing guidelines

### PR DESCRIPTION
**Pull request by Varunram**
w.r.t #3405, the contributing page has been improved, taking cue from Scala docs.

### CODE REVIEW

**Comment by allanrenucci** [comment]
Ping @Blaisorblade

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,63 +1,124 @@
...
+
+3. Bug Fix
+
+    At the end of the commit message, include "Fixes dotty/bug#NNNN", where https://github.com/lampepfl/dotty/issues/NNNN tracks the bug you're fixing. We also recommend naming your branch after the ticket number.
+
+4. Enhancement or New Feature
+
+    For longer-running development, likely required for this category of code contributions, we suggest you include `topic/` or `wip/` in your branch name, to indicate that this is work in progress, and that others should be prepared to rebase if they branch off your branch.
+
+    Any language change (including bug fixes) must be accompanied by the relevant updates to the spec, which lives in the same repository for this reason.
```
Comment (Blaisorblade) [inline CONTRIBUTING.md]: I think we don't have a fork of the spec in this repo? Maybe we should request language documentation in `docs/docs/`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,63 +1,124 @@
...
+
+5. Work In Progress
+
+    It is ok to work on a public feature branch in the GitHub repository. Something that can sometimes be useful for early feedback etc. If so, then it is preferable to name the branch accordingly. This can be done by either prefixing the name with ``wip-`` as in ‘Work In Progress’, or use hierarchical names like ``wip/..``, ``feature/..`` or ``topic/..``. Either way is fine as long as it is clear that it is work in progress and not ready for merge. This work can temporarily have a lower standard. However, to be merged into master it will have to go through the regular process outlined above, with Pull Request, review etc..
```
Comment (Blaisorblade) [inline CONTRIBUTING.md]: '..' should be '.' (also in the original)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,63 +1,124 @@
-# Dotty Developer Guidelines
+# Thank you for contributing to Dotty
 
-These guidelines are meant to be a living document that should be changed and adapted as needed. We encourage changes that make it easier to achieve our goals in an efficient way.
```
Comment (Blaisorblade) [inline CONTRIBUTING.md]: I think (not strongly) we should keep this paragraph since I feel it's still accurate.

**Comment by odersky** [comment]
This has been buried for a year. @Blaisorblade if you assign to yourself it is your responsibility to move this along or to assign to someone else. Adding @smarter for possible additional perspective.

**Comment by Blaisorblade** [comment]
Sorry about that. This was assigned to @smarter for a while before me. There seems to not be enough interest in this. So I'm closing this. (Feel free to reopen).

**Pull request closed without merging**